.github/workflows/linux.yml

```diff
@@ -0,0 +1,247 @@
+          make -C build -j$(nproc)
+          make -C build -j$(nproc) utrecht_tiles
+          make -C build -j$(nproc) tests gurka
+      
+      - name: Save cache
+        uses: actions/cache/save@v4
+        with:
+          path: /home/runner/.cache/ccache
+          key: ccache-release-linux-x86_64-${{ env.BRANCH_NAME }}-${{ env.TIMESTAMP }}
+
+      - name: Run tests
+        run: ASAN_OPTIONS="detect_leaks=0" make -C build -j$(nproc) check
+      
+      - name: Install Valhalla
+        run: |
+          sudo make -C build install
+          make -C build package
+      
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        if: ${{ env.SETUP_TMATE == 'true' || failure() }}
+        with:
+          detached: true
+
+  debug_build:
+    name: Debug build
+    runs-on: ubuntu-latest
+    needs: [lint_format]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 1
+      
+      - name: Install deps
+        run: ./scripts/install-linux-deps.sh
+
+      - name: Get timestamp & ccache dir
+        run: |
+          echo "TIMESTAMP=$(date +"%s")" >> $GITHUB_ENV
+          echo "CCACHE_DIR=$(ccache --get-config cache_dir)" >> $GITHUB_ENV
+
+      - name: Restore cache
+        id: cache-ccache-restore
+        uses: actions/cache/restore@v4
+        with:
+          path: /home/runner/.cache/ccache
+          key: ccache-debug-linux-x86_64-${{ env.BRANCH_NAME }}
+          restore-keys: |
+            ccache-debug-linux-x86_64-master
+            ccache-debug-linux-x86_64
+      
+      - name: Build Valhalla
+        run: |
+          # NOTE: -Werror disabled in CI, as we currently have >4k warnings.
+          cmake -B build -DCMAKE_BUILD_TYPE=Debug -DENABLE_COVERAGE=On -DCPACK_GENERATOR=DEB \
+            -DENABLE_COMPILER_WARNINGS=On -DENABLE_WERROR=Off -DCMAKE_EXPORT_COMPILE_COMMANDS=On \
+            -DCMAKE_CXX_FLAGS="-fuse-ld=lld" -DLOGGING_LEVEL=INFO -DENABLE_PYTHON_BINDINGS=On -DENABLE_GDAL=On \
+            -DBUILD_SHARED_LIBS=On
+          make -C build -j$(nproc)
+          make -C build -j$(nproc) utrecht_tiles
+          make -C build -j$(nproc) tests gurka
+      
+      - name: Save cache
+        uses: actions/cache/save@v4
+        with:
+          path: /home/runner/.cache/ccache
+          key: ccache-debug-linux-x86_64-${{ env.BRANCH_NAME }}-${{ env.TIMESTAMP }}
+      
+      - name: Run clang-tidy
+        run: scripts/clang-tidy-only-diff.sh $(nproc)
+
+      - name: Run tests
+        run: make -C build -j$(nproc) check
+      
+      - name: Install & Package Valhalla
+        run: |
+          df -h
+          sudo make -C build install
+          make -C build package
+
+      - uses: codecov/codecov-action@v5
+        name: Upload codecov report
+        with:
+          fail_ci_if_error: true
+          token: ${{ secrets.CODECOV_TOKEN }}
+          verbose: true
```

Comment:
as long as the debug build is fine we publish the codecov report. I think before that was only done when _all_ jobs were successful.